src/Aspire.Hosting/ExternalServiceBuilderExtensions.cs

```diff
@@ -104,7 +105,9 @@ private static IResourceBuilder<ExternalServiceResource> AddExternalServiceImpl(
                 if (uri is null)
                 {
                     // If the URI is not set, it means we are using a parameterized URL
-                    var url = resource.UrlParameter?.Value;
+                    var url = resource.UrlParameter is null
+                            ? null
+                            : await resource.UrlParameter.GetValueAsync(ct).ConfigureAwait(false);
```

Comment:
We want to handle any exception here and update the resource state to something like "FailedToStart" like the logic below yeah?

Reply:
Done